server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -660,7 +659,7 @@ public boolean advanceExact(int doc) throws IOException {
                 @Override
                 public double doubleValue() throws IOException {
                     final GeoPoint from = fromPoints[0];
-                    final GeoPoint to = singleValues.geoPointValue();
+                    final MultiGeoValues.GeoValue to = singleValues.nextValue();
```

Comment:
Should we assert `singleValues.docValueCount() == 1`?

Reply:
I guess that makes sense 👍